### DISCUSSION TITLE
Update hvPlot default categorical colormap to use glasbey_hv

### DISCUSSION THREAD

**Discussion by Azaya89** [holoviz/hvplot#1547] [General]
#### Is your feature request related to a problem? Please describe.

This is not a problem per se, but rather an observation which was raised on the discord channel: the default categorical colormaps in HoloViews and hvPlot do not match.

#### Describe the solution you’d like

I propose changing the default categorical colormap in hvPlot to `glasbey_hv`, which is currently used in HoloViews. My reasoning is as follows:

hvPlot is often promoted as a high-level, more user-friendly alternative to HoloViews. For users moving from HoloViews to hvPlot, it would be beneficial to maintain certain features that enhance the user experience. By using the same default categorical colormap, we can create a more seamless transition and avoid the need for users to manually adjust the colormap to preserve their familiar visual settings.

#### Describe alternatives you’ve considered

	1.	Continuing to use the current default colormap in hvPlot.
	2.	Manually importing the `glasbey_hv` colormap from colorcet to maintain consistency across HoloViews and hvPlot.

#### Additional context
Here's a visual representation of the differences:

```python
import pandas as pd

data = {'x': range(10),
 'y': range(10),
 'cat': ['cat1', 'cat2', 'cat3', 'cat4', 'cat5',
         'cat6', 'cat7', 'cat8', 'cat9', 'cat10']}
df = pd.DataFrame(data)

# Using hvplot
import hvplot.pandas # noqa

df.hvplot.points('x', 'y', by='cat')
```

![hvplot_color](https://github.com/user-attachments/assets/b1b7ce3b-0b0e-445d-952c-87b5ff02aa40)

```python
# using holoviews
import holoviews as hv
from holoviews.operation.datashader import datashade
import datashader as ds
hv.extension('bokeh')

plot = hv.Points(df, ["x", "y"], ['cat'])
datashade(plot, aggregator=ds.by("cat"))
```

![holoviews_color](https://github.com/user-attachments/assets/39f12719-c255-4047-92de-0f21cbc6db10)

**Comment by Azaya89** [comment]
Using holoviews without the `datashader` aggregator seems to use a continuous colormap as the default:

```python
hv.Points(df, ["x", "y"], ['cat']).opts(color='cat')
```
![holoview_sans_ds](https://github.com/user-attachments/assets/5d93356e-0695-4d16-8b2b-6ec376e3ba95)

**Comment by maximlt** [comment]
> I propose changing the default categorical colormap in hvPlot to `glasbey_hv`, which is currently used in HoloViews.

This is not exactly right I think, HoloViews seems to just cycle through a default set of colors (blue, red, gold, etc.) defined here:
https://github.com/holoviz/holoviews/blob/4b668d390f63ea5bdb9f96cad1933805ea6400fb/holoviews/plotting/__init__.py#L59-L60

```python
N = 20
data = {'x': range(N), 'y': range(N), 'cat': [f'cat{i}' for i in range(N)]}
df2 = pd.DataFrame(data)
points = hv.Points(df2, ["x", "y"], ['cat'])
points.groupby('cat', hv.NdOverlay).opts(hv.opts.Points(size=5)).opts(width=700, legend_position='right', legend_cols=2)
```

![image](https://github.com/user-attachments/assets/50a260a7-c0a5-44ab-953b-201b816a19b4)

> Using holoviews without the `datashader` aggregator seems to use a continuous colormap as the default:

Yep it looks like HoloViews just defaults to a linear colormap in this case, the [style mapping user guide](https://holoviews.org/user_guide/Style_Mapping.html#categorical-values) shows how to set a a categorical colormap in this case.

![image](https://github.com/user-attachments/assets/d9f9c9e8-05fc-4331-851e-e0acc5ab9223)

---

```python
import holoviews as hv
import hvplot.pandas

import pandas as pd
import numpy as np

N_POINTS_PER_GROUP = 1_000
GRID_DIM = 3
cat_k = 0
groups = []
for i in range(GRID_DIM):
    for j in range(GRID_DIM):
        groups.append((j, i, 0.25, f'cat{cat_k}'))
        cat_k += 1
dists = [
    pd.DataFrame(dict(x=np.random.normal(x, s, N_POINTS_PER_GROUP), y=np.random.normal(y, s, N_POINTS_PER_GROUP), cat=cat))
     for x,  y, s, cat in groups
]
df = pd.concat(dists, ignore_index=True)
```

* Only `by=<cat_col>`

```python
df.hvplot.points(by='cat').opts(legend_cols=2)
```

![image](https://github.com/user-attachments/assets/18428fff-69e0-4c97-8d73-c3b7d40f1ad3)

Internally, this is equivalent to:

```python
points = hv.Points(df, ["x", "y"], ['cat'])
points.groupby('cat', hv.NdOverlay).opts(hv.opts.Points(size=5)).opts(width=700, legend_position='right')
```

![image](https://github.com/user-attachments/assets/86a941be-953d-4530-9bfd-a0e9c75462b5)

The two plots above from hvPlot and HoloViews use the default HoloViews cycle.

* `by=<cat_col>` and `datashade=True`

```python
df.hvplot.points(by='cat', datashade=True, dynspread=True, threshold=1)
```

![image](https://github.com/user-attachments/assets/13e2dae7-750d-4af0-8e10-60fb2893c99e)

hvPlot uses `glasbey_category10`.

* `color=<cat_col>`

```python
df.hvplot.points(color='cat').opts(legend_cols=8)  # had to set a weird value for legend_cols...
```

![image](https://github.com/user-attachments/assets/26714460-bdd0-4af4-b0f0-328b40d14c6a)


hvPlot uses `glasbey_category10`, while, as already mentioned, HoloViews defaults to the linear blues cmap:

```python
hv.Points(df, ["x", "y"], ['cat']).opts(color='cat', legend_position='right', width=700, size=5, legend_cols=8)
```

![image](https://github.com/user-attachments/assets/358b5b5a-210d-4ce6-ba89-95cbd7ddf0ea)

The hvPlot styling can be reproduced with:

```python
import colorcet as cc
hv.Points(df, ["x", "y"], ['cat']).opts(color='cat', legend_position='right', width=700, size=5, cmap=cc.palette['glasbey_category10'], legend_cols=8)
```

![image](https://github.com/user-attachments/assets/bb4c2305-636d-46f8-b954-29040be8e2f0)

---

The two categorical colormaps `glasbey_hv` and `glasbey_category10` can easily be compared on colorcet's [website](https://colorcet.holoviz.org/user_guide/Categorical.html#starting-colors):

![image](https://github.com/user-attachments/assets/013e236e-7662-4299-b542-37e608af1d96)

❓ In fact, I don't know if the base color cycle of HoloViews is ad-hoc or comes from a reference color set?

For reference, `glasbey_hv` was added in https://github.com/holoviz/colorcet/pull/25 (March 2019), see the discussion from https://github.com/holoviz/colorcet/issues/11. hvPlot seems to have been using `Category10` since its creation and then `glasbey_category10` when it was added to `colorcet`.

**Comment by maximlt** [comment]
From the above:

- hvPlot does not use the same cmap when `by` or `color` are set, this should be fixed imo.
- I'm not sure the argument to move to `glasbey_hv` is strong enough, it's quite an annoying change for users so we'd really have to motivate it. Sure, making the life of HoloViews users would be nice! But hvPlot is designed as a replacement of the Pandas `.plot` API (and Xarray, etc.) and I'd rather make the life of users coming from these libraries easier (there are many more of them than HoloViews users!). Let's note that this discussion followed the attempt to update the OpenSky example on examples.holoviz.org (https://github.com/holoviz-topics/examples/pull/396) from using HoloViews'API to hvPlot's API and, yes, in this kind of exercise the difference in cmap was noticeable and disturbing.

**Comment by Azaya89** [comment]
Thank you for the clarification @maximlt. However, there is a sort of disconnect in my head that happens when you do

```python
points = hv.Points(df, ['x', 'y'])
points.opts(color='cat')
```
for example and get a plot like this:

![holoviews_cat](https://github.com/user-attachments/assets/d19348c4-f543-4e25-a39d-2a433b24e782)

versus when you do:
```python
points.opts(color='cat', cmap='glasbey_hv')
```

which is equivalent to
```python
points.groupby('cat', hv.NdOverlay)
```
and get a plot like this:

![holoviews_color_cat](https://github.com/user-attachments/assets/e06b25db-33aa-4f57-b096-51c69b876ea9)


This seems to imply (I don't know if I'm correct) that the `glasbey_hv` color mapping will only apply by default to groups of `NdOverlay` that have been...'overlaid' together. This is not intuitive for new users and may cause some confusion on what to do if they get the linear colormapping on first try without specifying the `cmap` argument.
